src/tests/class_tests/openms/source/IDFilter_test.cpp

```diff
@@ -180,6 +182,52 @@ START_SECTION((static void extractPeptideSequences(const vector<PeptideIdentific
 }
 END_SECTION
 
+START_SECTION((class PeptideDigestionFilter::operator(PeptideHit& hit)))
+{
+  EnzymaticDigestion digestion;
+  digestion.setEnzyme("Trypsin");
+  
+  IDFilter::PeptideDigestionFilter filter(digestion, 0, 1);
+  vector<PeptideHit>hits, test_hits;
+
+  
+  // No cleavage
+  hits.push_back(PeptideHit(0, 0, 0, AASequence::fromString("(MOD:00051)DFPIANGER")));
+  hits.push_back(PeptideHit(0, 0, 0, AASequence::fromString("DFPIANGER")));
+  hits.push_back(PeptideHit(0, 0, 0, AASequence::fromString("DFPIAN(Deamidated)GER")));
```

Comment:
I find difficult to be certain of such example.
If I use Asp-N_ambic `Asp-N Ammonium bicarbonate cleaves before D or E.`
and `F(MOD:00051)DFPIANGR` as the sequence will that satisfy the case you re describing (zero missed cleavages)?

Reply:
yes. perfect. thanks.

Reply:
It bit me for the sequence `"F(MOD:00051)DFPIANGER"` 

`Message: the value 'MOD:00051' was used but is not valid; Retrieving the modification failed. It is not available for the residue 'F' and term specificity 0.`

Reply:
`MOD:00051` = `A protein modification that effectively converts an L-aspartic acid residue to N-acetyl-L-aspartic acid.` As I understand this modification can only occur at aspartic acid  `D` and not `F`.

Where is the  sequence `F(MOD:00051)DFPIANGER` from?

Reply:
The isue of cleavages after modifiied amino acids is very complicated. It depends on the enzyme and the modification, for example: Trypsin can cleave after monomethylated Lysine, but not dimethylated or acetylated lysines.

Reply:
Currently digestion works naively with UnmodifiedStrings. This seems to be a functionality missing in digestion, that is ignoring modifications that may nullify the rule for a certain cleavage.

Reply:
@cbielow Are you fine with the sequence `K(Dimethyl)FPIAUGR` for the test?

Reply:
yes. In combination with enyzme `Asp-N_ambic` (Asp-N Ammonium bicarbonate cleaves before D or E.) this would test exactly what was intended, i.e.
a cleavage before `D` should not occur (since its part of the modification name, which should not be passed into the regex).